### PR TITLE
BIGNUM signed add/sub routines refactory

### PR DESCRIPTION
**Pull request by davxy**
Old code replaced in favor of a clearer implementation.
Performances are not penalized.

Updated the copyright end date to 2018.

The new implementation succeeded with the provided BIGNUM unit tests and is functionally equivalent
to the previous one, only a bit cleaner.

##### Checklist

- [ ] documentation is added or updated
- [ ] tests are added or updated

### CODE REVIEW

**Comment by richsalz** [comment]
Close/open to kick the CLA bot.

**Comment by richsalz** [comment]
This looks okay to me, but I'd really like @davidben or @dot-asm to take a look.

File: crypto/bn/bn_add.c [hunk 1]
```diff
@@ -10,51 +10,69 @@
...
+    if (a->neg != b->neg) {
+        r_neg = a->neg;
+        ret = BN_uadd(r, a, b);
+    } else {
+        cmp_res = BN_ucmp(a, b);
+        if (cmp_res > 0) {
+            r_neg = a->neg;
+            ret = BN_usub(r, a, b);
+        } else if (cmp_res < 0) {
+            r_neg = 1 - (b->neg != 0);
```
Comment (davidben) [inline crypto/bn/bn_add.c]: Why not just `r_neg = !b->neg`?

Comment (davxy) [reply crypto/bn/bn_add.c]: Looks a lot better ;-)
I'll change it asap!

**Comment by davxy** [comment]
Update pushed

**Comment by richsalz** [comment]
Merged to master, thank you!

**Pull request closed without merging**